Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -161,11 +161,13 @@ void WebCookieJar::setCookies(WebCore::Document& document, const URL& url, const
     if (!webFrame || !webFrame->page())
         return;
 
+    auto sameSiteInfo = CookieJar::sameSiteInfo(document, IsForDOMCookieAccess::Yes);
     ApplyTrackingPrevention applyTrackingPreventionInNetworkProcess = ApplyTrackingPrevention::No;
-    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), url, applyTrackingPreventionInNetworkProcess))
+    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), url, applyTrackingPreventionInNetworkProcess)) {
+        setCookiesInPartitionedCookieStorage(document, url, sameSiteInfo, cookieString);
```

Comment:
Maybe we can add the `IfNecessary` suffix here to clarify that it doesn't always happen. As a matter of fact, it rarely happens.